src/test/unit/controls/Line/LineLoad-spec.js

```diff
@@ -1226,6 +1226,91 @@ describe("Line - Load", () => {
                 done();
             });
         });
+        describe("if the legend has no data and user sets showElement to be false", () => {
+            it("should hide the legend", () => {
+                const input = getInput([], false, false);
+                input.legendOptions = {
+                    showElement: false
+                }
+                graphDefault.loadContent(new Line(input));
+                const legendContainer = fetchElementByClass(
+                    lineGraphContainer,
+                    styles.legendItem
+                );
+                expect(legendContainer).not.toBeNull();
+                expect(legendContainer.tagName).toBe("LI");
+                const legendItem = document.body.querySelector(
+                    `.${styles.legendItem}`
+                );
+                expect(legendItem.getAttribute("style")).toBe("display: none; padding: 4px 8px;");
+            })
+        });
+        describe("if legend has no data and user sets showElement to be true", () => {
+            it("should show the legend", () => {
+                const input = getInput([], false, false);
+                input.legendOptions = {
+                    showElement: true,
+                }
+                graphDefault.loadContent(new Line(input));
+                const legendContainer = fetchElementByClass(
+                    lineGraphContainer,
+                    styles.legendItem
+                );
+                const legendItems = legendContainer.children;
+                expect(legendContainer).not.toBeNull();
+                expect(legendContainer.tagName).toBe("LI");
+                expect(legendItems.length).toBe(2);
+                const legendItem = document.body.querySelector(
+                    `.${styles.legendItem}`
+                );
+                expect(legendItem.getAttribute("aria-disabled")).toBe("true");
+                expect(legendItem.getAttribute("aria-current")).toBe("true");
+            })
+        });
```

Comment:
Describe blocks are meant to group together similar tests. These tests can be combined into a single describe block.

```js
describe("when the legend has no data",  ()=>{
   it ("should hide the legend if showElement is false", ()=>{....})
        ......
   it ("should show the legend if showElement is true", ()=>{....})
        ......
}
```

Reply:
Updated [here](https://github.com/NarasimhaShenoi/carbon-graphs/commit/e5eae0edb49ef09342b04de3df0500d909b9e134)